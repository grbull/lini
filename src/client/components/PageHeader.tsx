import cn from 'classnames';
import React, { ReactElement, ReactNode } from 'react';
import { Helmet } from 'react-helmet';

import { BackButton } from './BackButton';

interface Props {
  className?: string;
  goBack?: boolean;
  isLoading?: boolean;
  title: string;
  rightElement?: ReactNode;
}

export function PageHeader({
  className,
  goBack,
  isLoading,
  title,
  rightElement,
}: Props): ReactElement {
  return (
    <>
      <Helmet>
        <title>{title} - Lini</title>
      </Helmet>
      <header
        className={cn(
          'w-full',
          'h-12',
          'flex',
          'items-center',
          'p-2',
          'md:hidden',
          { 'justify-between': goBack, 'justify-center': !goBack },
          className
        )}
      >
        {goBack && <BackButton />}
        {isLoading ? (
          <div
            className={cn('w-2/3', 'h-3/4', 'animate-pulse', 'bg-gray-600')}
          />
        ) : (
          <h1
            className={cn(
              'mx-2',
              'font-medium',
              'text-lg',
              'overflow-x-auto',
              'whitespace-nowrap'
            )}
          >
            {title}
          </h1>
        )}
        {goBack && <div className={cn('w-11')}>{rightElement}</div>}
      </header>
    </>
  );
}

PageHeader.defaultProps = {
  className: undefined,
  goBack: true,
  isLoading: false,
  rightElement: undefined,
};
