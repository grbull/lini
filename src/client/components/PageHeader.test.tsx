/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom/extend-expect';

import { render, RenderResult } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import React from 'react';
import { Router } from 'react-router-dom';

import { PageHeader } from './PageHeader';

function setup({ ...props }: Record<string, unknown>): RenderResult {
  const history = createMemoryHistory();

  const utils = render(
    <Router history={history}>
      {/* eslint-disable-next-line react/jsx-props-no-spreading */}
      <PageHeader title="Testing" {...props} />
    </Router>
  );

  return { ...utils };
}

describe('PageHeader Component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = setup({});
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches the snapshot when loading', () => {
    const { asFragment } = setup({ isLoading: true });
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches the snapshot with a titleLink', () => {
    const { asFragment } = setup({ titleLink: '/' });
    expect(asFragment()).toMatchSnapshot();
  });
});
