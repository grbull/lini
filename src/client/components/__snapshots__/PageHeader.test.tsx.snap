// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader Component matches the snapshot 1`] = `
<DocumentFragment>
  <header
    class="w-full h-12 flex items-center p-2 md:hidden justify-between"
  >
    <button
      class="w-11 float-left"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-arrow-left fa-w-14 m-1 text-lg"
        data-icon="arrow-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
          fill="currentColor"
        />
      </svg>
    </button>
    <h1
      class="mx-2 font-medium text-lg overflow-x-auto whitespace-nowrap"
    >
      Testing
    </h1>
    <div
      class="w-11"
    />
  </header>
</DocumentFragment>
`;

exports[`PageHeader Component matches the snapshot when loading 1`] = `
<DocumentFragment>
  <header
    class="w-full h-12 flex items-center p-2 md:hidden justify-between"
  >
    <button
      class="w-11 float-left"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-arrow-left fa-w-14 m-1 text-lg"
        data-icon="arrow-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
          fill="currentColor"
        />
      </svg>
    </button>
    <div
      class="w-2/3 h-3/4 animate-pulse bg-gray-600"
    />
    <div
      class="w-11"
    />
  </header>
</DocumentFragment>
`;

exports[`PageHeader Component matches the snapshot with a titleLink 1`] = `
<DocumentFragment>
  <header
    class="w-full h-12 flex items-center p-2 md:hidden justify-between"
  >
    <button
      class="w-11 float-left"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-arrow-left fa-w-14 m-1 text-lg"
        data-icon="arrow-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
          fill="currentColor"
        />
      </svg>
    </button>
    <a
      class="mx-2 font-medium text-lg overflow-x-auto whitespace-nowrap"
      href="/"
    >
      Testing
    </a>
    <div
      class="w-11"
    />
  </header>
</DocumentFragment>
`;
